### DISCUSSION TITLE
🦕 Upcoming 0.3.0-alpha release 🦕

### DISCUSSION THREAD

**Discussion by w4ffl35** [w4ffl35/krita_stable_diffusion#24] [Announcements]
I am working on version 0.3.0 of the Krita Stable Diffusion plugin.

**Planned features**

- 🦕 **Installation script for Linux**:  no more messy setup
- 🦕 **Streamlined builds**: allows you to get updates faster
- 🦕 **Windows build with installation script** - run this plugin in Krita on windows

Please keep an eye on this thread where I we can discuss the update and any bugs we might encounter.

**Comment by w4ffl35** [comment]
![image](https://user-images.githubusercontent.com/25737761/191878771-c0e301fe-4017-48d5-a6ab-6757b4653e1b.png)

# Approximate Progress towards 0.3.0-alpha

**Installation script for Linux: no more messy setup**
█████████▒ 90%  **Estimated time remaining:** 1 hour

**Streamlined builds: allows you to get updates faster**
█████████▒ 90%  **Estimated time remaining:** 2 hours

**Windows build with installation script - run this plugin in Krita on window**
█▒▒▒▒▒▒▒▒▒ 10%  **Estimated time remaining:** 1 to 8 hours

---

### Linux installation script

There is less than an hour left on this task. I just need to do some final tests.

### Development environment

TL;DR I have about 1 or 2 hours of setup time remaining and then I will be able to release faster.

---

Here's what I've got so far, why its important, and where I need to be before I can release v0.3.0-alpha

![image](https://user-images.githubusercontent.com/25737761/191879075-44a5ee01-2ea3-4ee5-aafb-6c7a3e88c9e6.png)

**WARNING its about to get technical**

In order to quickly release new features fix bugs it is important to have several things in place

- An excellent development environment 
- A CI/CD pipeline that won't break
- The ability to setup my development environment with a single command
- The ability to switch between multiple builds and environments quickly and efficiently

After spending quite a bit of time working through various issues, now I have those things. Here's what that means in a little more detail:

- [x] A Docker image + Dockerfile which creates dev and production environments
- [x] Automated build scripts which generate dev, production and test builds of the plugin and stable diffusion
- [x] An installation script which deflates and moves the required files to the correct destination
- [x] An easy way to switch between environments and launch Krita so that i can quickly switch between production, feature etc. environments
- [x] One line setup / teardown scripts
- [x] Ability to skip portions of packaging process with flags
- [x] Everything in bash scripts

---

### Windows support

I have a plan in place for Windows support that I have yet to test. This could end up taking anywhere from an hour to another day or two.

Join my discord to see my status, ask me direct questions and get updates sooner as the happen.

[![Discord](https://img.shields.io/discord/839511291466219541?color=5865F2&logo=discord&logoColor=white&style=for-the-badge)](https://discord.com/channels/839511291466219541/1022298294338191381)

**Comment by w4ffl35** [comment]
Final stages now. Just wrapping up the ci/cd pipeine and testing, then testing install script, then moving on to windows support.

Then we can finally start getting some real features added while people play around with image generation.

**Comment by w4ffl35** [comment]
# Approximate Progress towards 0.3.0-alpha

**Streamlined builds: allows you to get updates faster**
██████████ 95%  **Estimated time remaining:** 4 hours

**Windows build with installation script - run this plugin in Krita on window**
█████▒▒▒▒▒ 50%  **Estimated time remaining:** 4 hours with testing

**Complete**
**Installation script for Linux: no more messy setup**
**██████████ 100%**

![programmerhardatworkprogramming](https://user-images.githubusercontent.com/25737761/192125437-308feec4-81f3-4bc8-be28-18b56051d682.png)

I have been hard at work addressing issues in preparation for [0.2.1-alpha](https://github.com/w4ffl35/krita_stable_diffusion/releases/tag/0.2.1) which finally launched today.

As promised this version includes an installation script, and should fix several bugs that people have reported.

**Comment by w4ffl35** [comment]
As I write this update, version 0.2.2-alpha is finalizing (tar zipping the package and uploading to AWS). In approximately 60 minutes or less it will be released and available for you to download.

Thanks to the bug reports of users and several more in Discord, I have discovered a large number of issues with the install script, including missing models, issues with missing classes and more.

![image](https://user-images.githubusercontent.com/25737761/192170700-e8931a90-2718-4551-be69-85171ff0e2e9.png)

Since release I was most likely the only person with access to a working copy of this plugin. 

For this crime, I was promptly fired from the project.

![image](https://user-images.githubusercontent.com/25737761/192170775-adb9ff73-bf3e-4742-9415-972d11c51e64.png)

But I rehired myself, because I can.

![image](https://user-images.githubusercontent.com/25737761/192170982-90de755c-b86f-4263-b482-b1e92223fd49.png)

I'm here to tech support you

![image](https://user-images.githubusercontent.com/25737761/192170857-564537c7-53d7-4a33-a958-8cd1e11625a4.png)

... so don't turn your back on Krita Stable Diffusion
